research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,324 @@
+  We have various mechanisms in place to guarantee confidentiality.
+  The strongest option are secure channels setup with the help of the [Noise Protocol Framework](https://noiseprotocol.org/).
+* meta-data protection, 
+  - TODO: what meta-data (outside of anonymity related) should be protected by the privacy-preserving property
+
+### Anonymity
+
+Privacy and anonymity are closely linked.
+Both the identity of a user and data that allows inferring a user's identity should be part of the privacy policy.
+For the purpose of analysis, we want to have a clearer separation between these concepts.
+
+We define anonymity as *unlinkablity of users' identities and their shared data and/or actions*.
+
+We subdivide anonymity into *receiver anonymity* and *sender anonymity*.
+
+#### Receiver Anonymity
+
+We define receiver anonymity as *unlinkability of users' identities and the data they receive and/or related actions*.
+The data transmitted via Waku relay must be a [Waku message](https://rfc.vac.dev/spec/14/), which contains a content topic field.
+Because each topic is associated with a content topic, and each receiver is interested in messages with specific content topics,
+receiver anonymity in the context of Waku corresponds to *subscriber-topic unlinkability*.
+An example for the "action" part of our receiver anonymity definition is subscribing to a specific topic.
+
+The Waku message's content topic is not related to the libp2p pubsub topic.
+For now, Waku uses a single libp2p pubsub topic, which means messages are propagated via a single mesh of peers.
+With this, the receiver discloses its participation in Waku on the gossipsub layer.
+We will leave the analysis of libp2p gossipsub to a future article within this series, and only provide a few hints and pointers here.
+
+Waku offers k-anonymity regarding content topic interest in the global adversary model.
+Assuming there are $n$ Waku content topics, a receiver has $n$-anonymity with respect to association to a specific content topic.
+In the future, Waku might distribute messages over several libp2p pubsub topics.
+This would still yield $k$-anonymity, assuming $k$ content topics share the same pubsub topic.
+Waku is not directly concerned with 1:1 communication, so for this post, 1:1 communication is out of scope.
+A 1:1 communication protocol might be added to Waku in the future.
+It would maintain receiver anonymity leveraging [partitioned topics](https://specs.status.im/spec/10#partitioned-topic).
+
+#### Sender Anonymity
+
+We define sender anonymity as *unlinkability of users' identities and the data they send and/or related actions*.
+Because the data in the context of Waku is Waku messages, sender anonymity corresponds to *sender-message unlinkability*.
+
+In summary, Waku offers sender anonymity in the single node and local attacker models because of [Waku's strict no sign policy](https://rfc.vac.dev/spec/11/#signature-policy).
```

Comment:
I'd give some cred to libp2p gossipsub spec work here, this came out of Ethereum CL p2p spec work

https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#why-are-we-using-the-strictnosign-signature-policy

Reply:
I was not aware. Added in dc0c7c77b